File: src/geonhyeong/week8/튜플.swift
/*
*  튜플.swift
*  Question Link: https://school.programmers.co.kr/learn/courses/30/lessons/64065
*  Primary idea:   튜플
*                   1. 
*
*  걸린 시간 : 15:00
*  Time Complexity : O()
*  Space Complexity : O()
*  Runtime:  ms
*  Memory Usage:  MB
*/

import Foundation

func solution(_ s:String) -> [Int] {
    var result = [Int]()
    let tuples = s.components(separatedBy: ["{","}"]).filter({$0 != "" && $0 != ","})
    let sortedTuples = tuples.sorted{$0.count < $1.count} // 갯수로 오름차순

    for tuple in sortedTuples {
        let removeComma = tuple.split(separator: ",").map{Int($0)!}
        
        for element in removeComma {
            if !result.contains(element) {
                result.append(element)
            }
        }
    }
    
    return result
}

File: src/jimin/week8/튜플.py
def solution(s):
    answer = []

    tuple_list = []
    num = ""
    flag = False
    for element in s[1:-1]:
        if element == '{':
            tuple_element = []
            flag = True
        if element == '}':
            tuple_element.append(int(num))
            tuple_list.append(tuple_element)
            num = ""
            flag = False
        if element == ',' and flag:
            tuple_element.append(int(num))
            num = ""
        if element.isdigit():
            num += element

    tuple_list = sorted(tuple_list, key=lambda x: len(x))

    for tuple in tuple_list:
        for i in tuple:
            if not i in answer:
                answer.append(i)
    return answer

s = "{{20,111},{111}}"
print(solution(s))